### PR TITLE
Draft support for hand tracking

### PR DESCRIPTION
**Pull request by Manishearth**
I wrote some support for hand tracking using updated microsoft XML files. I wasn't quite sure how you'd want to structure this, features-wise, so I did it unconditionally as an example.

@Ralith would you like to take this PR and make it feature-conditional in the way you would like?

### CODE REVIEW

File: generator/src/main.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ fn main() {
     args.next().unwrap();
     let mut source = File::open(args.next().map(PathBuf::from).unwrap_or_else(|| {
         Path::new(env!("CARGO_MANIFEST_DIR"))
-            .join("../sys/OpenXR-SDK/specification/registry/xr.xml")
+            .join("xr.xml")
```
Comment (Ralith) [inline generator/src/main.rs:28]: The commit that adds the new files should be squashed into this one to make it practical to review the changes in the generated output.

**Comment by Ralith** [comment]
Sure, I can try hacking this up a bit. May not get to it for a few days though.

**Comment by Ralith** [comment]
My initial thought is that we can probably modify the generator to feature-guard a blacklisted set of extensions, which will make a good foundation.

File: generator/src/main.rs (hunk 1 above)
Comment (Manishearth) [reply generator/src/main.rs:28]: it is?

**Comment by Manishearth** [comment]
Sounds good, I'm not in a particular rush. I managed to get this working and I mostly just wanted to have a proof of concept.

Blacklisting extensions could work, though we need to blacklist all of the associated types as well.

I was thinking we have two sets of generated files, one for the regular xr.xml and one for the Microsoft one (They'll have it in their repo as well soon, I think), and feature flags to flip between the two. `openxr` also has feature flags that toggle the hand tracking (and future other extension) types in tandem

**Comment by Manishearth** [comment]
@Ralith any updates? if we can get the conditional compilation set up I can start working on the external APIs for most of the extension features :smile:

**Comment by Ralith** [comment]
Been a bit busy, but it's still on my list!

**Comment by Manishearth** [comment]
Good to hear!

In the meantime we're going to use a forked version, but hope to revert back once all the patches land.

**Comment by Ralith** [comment]
Folded into #30. Please review!

**Pull request closed without merging**